packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -192,13 +211,19 @@ export abstract class Action extends cdk.Construct {
     }
   }
 
-  protected addOutputArtifact(name: string): Artifact {
-    const artifact = new Artifact(this, name);
+  protected addOutputArtifact(name?: string): Artifact {
```

Comment:
You can instead do:
```ts
protected addOutputArtifact(name: string = this.stage._generateOutputArtifactName(this)) {
  const artifact = new Artifact(this, outputArtifactName);
  retenu artifact;
}
```

Reply:
I don't think that will work. I want to call this method from the Action subclasses with `props.outputArtifactName`, where `outputArtifactName` is an optional `string`.

Reply:
I think @RomainMuller is right, you should be able to do that. If you pass `undefined` to `name` (which will be the case if `props.outputArtifactName` is not defined, it will apply the default.

Reply:
I stand corrected :) will change.